RJL.UIP.HW3.ConditionalsMethods/Program.cs

```diff
@@ -0,0 +1,245 @@
+                    inputArray[j] = tempItemValue;
+                    i++;
+                    j--;
+                }
+            }
+            if (j > firstIndex) QuickSortArray(inputArray, firstIndex, j);
+            if (i < lastIndex) QuickSortArray(inputArray, i, lastIndex);
+        }
+
+        static void PrintArray(int[] inputArrayPrint)
+        {
+            foreach (int item in inputArrayPrint)
+            {
+                Console.Write(" " + item);
+            }
+        }
+        static int[] CreateRandomArray()
+        {
+            Random randomizer = new Random();
+            int randomArrayLength = randomizer.Next(20, 50);
+            int[] randomArray = new int[randomArrayLength];
+            for (int i = 0; i < randomArrayLength; i++)
+            {
+                randomArray[i] = randomizer.Next(0, 9);
+            }
+            return randomArray;
+        }
+        static void SetNullMinMaxArray(int[] inputArray)
+        {
+            int maxValueIndex = 0;
+            int minValueIndex = 0;
+            int tempIndex;
+            for (int i = 0; i < inputArray.Length - 1; i++)
+            {
+                minValueIndex = (inputArray[i + 1] >= inputArray[minValueIndex]) ? minValueIndex : i + 1;
+                maxValueIndex = (inputArray[i + 1] > inputArray[maxValueIndex]) ? i + 1 : maxValueIndex;
+            }
+            if (minValueIndex > maxValueIndex)
+            {
+                tempIndex = minValueIndex;
+                minValueIndex = maxValueIndex;
+                maxValueIndex = tempIndex;
+            }
+            for (int i = minValueIndex + 1; i < maxValueIndex; i++)
+            {
+                inputArray[i] = 0;
+            }
+        }
+        static string[] GetMultiplyTable(int countPeople)
+        {
+            Random randomizer = new Random();
+            int firstValueTable = 0;
+            int secondValueTable = 0;
+            int resultMultiply = 0;
+            string[] resultStringMultiplyArray = new string[countPeople];
+            string[] tempStringMultiplyArray = new string[countPeople];
+            for (int i = 0; i < countPeople; i++)
+            {
+                bool compareArrayItem = false;
+                do
+                {
+                    firstValueTable = randomizer.Next(1, 9);
+                    secondValueTable = randomizer.Next(1, 9);
+                    resultMultiply = firstValueTable * secondValueTable;
+                    resultStringMultiplyArray[i] = firstValueTable + "*" + secondValueTable + "=" + resultMultiply;
+                    tempStringMultiplyArray[i] = secondValueTable + "*" + firstValueTable + "=" + resultMultiply;
+
+                    for (int j = 0; j < countPeople; j++)
+                    {
+                        compareArrayItem = ((resultStringMultiplyArray[i] == resultStringMultiplyArray[j]) || (tempStringMultiplyArray[i] == resultStringMultiplyArray[j])) && i != j;
```

Comment:
I'm not sure but I suggest you should reassign **compareArrayItem** only if **compareArrayItem** is _false_. think about this